x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -1341,12 +1374,28 @@ public static void registerSettings(List<Setting<?>> settings) {
         private final Predicate<String> ignoreRealmsPredicate;
         private final Predicate<String> ignoreRolesPredicate;
         private final Predicate<String> ignoreIndicesPredicate;
+        private final Predicate<String> ignoreIndexPrivilegesPredicate;
+        private final Predicate<String> ignoreClusterPrivilegesPredicate;
+
+        private final ConcurrentHashMap<String, Collection<String>> indexActions = new ConcurrentHashMap<>();
+        private final ConcurrentHashMap<String, Collection<String>> clusterActions = new ConcurrentHashMap<>();
 
         EventFilterPolicy(String name, Settings settings) {
             this(name, parsePredicate(FILTER_POLICY_IGNORE_PRINCIPALS.getConcreteSettingForNamespace(name).get(settings)),
                     parsePredicate(FILTER_POLICY_IGNORE_REALMS.getConcreteSettingForNamespace(name).get(settings)),
                     parsePredicate(FILTER_POLICY_IGNORE_ROLES.getConcreteSettingForNamespace(name).get(settings)),
-                    parsePredicate(FILTER_POLICY_IGNORE_INDICES.getConcreteSettingForNamespace(name).get(settings)));
+                    parsePredicate(FILTER_POLICY_IGNORE_INDICES.getConcreteSettingForNamespace(name).get(settings)),
+                    parsePredicate(FILTER_POLICY_IGNORE_INDEX_PRIVILEGES.getConcreteSettingForNamespace(name).get(settings)),
+                    parsePredicate(FILTER_POLICY_IGNORE_CLUSTER_PRIVILEGES.getConcreteSettingForNamespace(name).get(settings)));
+
+            if (FILTER_POLICY_IGNORE_INDEX_PRIVILEGES.getConcreteSettingForNamespace(name).exists(settings) &&
+                FILTER_POLICY_IGNORE_CLUSTER_PRIVILEGES.getConcreteSettingForNamespace(name).exists(settings)) {
+                final String message = String.format(
+                    Locale.ROOT,
+                    "Both Index and Cluster privilege ignore filters are set for policy [%s]. " +
+                        "Please update your configuration settings and remove one of these filters to avoid ambiguity.", this.name);
+                throw new IllegalArgumentException(message);
```

Comment:
We would need validation for the policies on node startup as well.
I remember @colings86 being a proponent of preventing the node from starting if validation fails.
This sounds reasonable, but I think it's still worth a bit of investigation and maybe socializing the direction more broadly.

In this particular case, the settings are dynamically update-able. So, if they're invalid we can set them via the persistent cluster settings API. I think it's worth going over the existing dynamic cluster settings that have validation, as well as over the settings that have validation when the node starts (I have not done so myself, maybe there's nothing interesting), and then maybe confirming the approach across the ES team.

Reply:
Dynamic updates which violate this rule will fail with appropriate messages now. If it happens at startup - node won't start